google-cloud-tools-plugin/testSrc/com/google/cloud/tools/intellij/appengine/facet/flexible/AppEngineFlexibleFrameworkDetectorTest.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.appengine.facet.flexible;
+
+import com.intellij.mock.MockVirtualFile;
+import com.intellij.openapi.fileTypes.FileTypeRegistry;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.patterns.ElementPattern;
+import com.intellij.util.indexing.FileContent;
+import com.intellij.util.indexing.FileContentImpl;
+import org.jetbrains.yaml.YAMLFileType;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/** Tests {@link AppEngineFlexibleFrameworkDetector} for valid/invalid App Engine Flex detection. */
+public class AppEngineFlexibleFrameworkDetectorTest {
+
+  private final String validAppEngineFlexYamlString =
+      AppEngineFlexibleFrameworkDetector.APP_ENGINE_REQUIRED_YAML + " java";
+
+  @Before
+  public void setUp() {
+    FileTypeRegistry mockFileTypeRegistry = mock(FileTypeRegistry.class);
+    // mock file type check routine. getInstance() is static, has to replace it with IJ Getter mock.
+    FileTypeRegistry.ourInstanceGetter = () -> mockFileTypeRegistry;
+    when(mockFileTypeRegistry.getFileTypeByFile(any(VirtualFile.class)))
+        .thenReturn(YAMLFileType.YML);
+  }
+
+  @Test
+  public void testIncompleteAppYamlDetection() {
+    AppEngineFlexibleFrameworkDetector detector = new AppEngineFlexibleFrameworkDetector();
```

Comment:
This should be made an instance variable to remove bloat from each test method (since this is the class under test so you'll always need an instance of it).